pkg/k8s/policyhelper/policy_test.go

```diff
@@ -16,7 +16,7 @@ import (
 	"github.com/aws/aws-application-networking-k8s/pkg/model/core"
 )
 
-func Test_getAttachedPolicy(t *testing.T) {
+func Test_GetAttachedPolicies(t *testing.T) {
 	type args struct {
```

Comment:
Could you extend the `Test_GetAttachedPolicies` to make it cover the code path that one obj have multiple different same kind policies? 

IIRC, in our CRDs for example: targetgrouppolicy_types.go vpcassociationpolicy_types.go

The `GetItems()` has a bug, this one is wrong:
```
func (pl *VpcAssociationPolicyList) GetItems() []core.Policy {
	items := make([]core.Policy, len(pl.Items))
	for i, item := range pl.Items {
		items[i] = &item
	}
	return items
}
```

it should change to this one:
```
func (pl *VpcAssociationPolicyList) GetItems() []core.Policy {
	items := make([]core.Policy, len(pl.Items))
	for i := range pl.Items {
		items[i] = &pl.Items[i]
	}
	return items
}

```

I think you need to do that change in this PR as well?

if you make `Test_GetAttachedPolicies` covers one obj having mutiple Policies, it could detect that bug.